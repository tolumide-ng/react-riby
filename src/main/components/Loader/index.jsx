import React from 'react';
import './index.scss';

const Loader = ({ className }) => {
  const loadingText =
    className === 'loader-small' ? '' : 'Loading...';

  return (
    <div className={`loader ${className}`}>
      <div className="loader-text">{loadingText}</div>
      <div
        data-testid="loader"
        className="loader-sector loader-sector-green"
      />
      <div className="loader-sector loader-sector-grey" />
      <div className="loader-sector loader-sector-dark-grey" />
    </div>
  );
};

export default Loader;
