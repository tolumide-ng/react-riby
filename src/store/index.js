import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './modules';

const middlewares = applyMiddleware(ReduxThunk);

const store = createStore(rootReducer, composeWithDevTools);

export default store;
