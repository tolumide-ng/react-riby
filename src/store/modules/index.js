import { combineReducers } from 'redux';
import allGifReducer from './allGifs/reducers';
import detailedGifReducer from './detailedGif/reducers';
import searchGifReducer from './searchGif/reducers';

const rootReducer = combineReducers({
  allGifReducer,
  detailedGifReducer,
  searchGifReducer,
});

export default rootReducer;
