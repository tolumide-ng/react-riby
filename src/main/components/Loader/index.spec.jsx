/* eslint-disable react/jsx-props-no-spreading */
import { render, fireEvent, cleanup } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Loader from './index';
import store from '../../../store';

describe('Loader', () => {
  test('Should return the Loader', () => {
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Loader />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
    expect(getByTestId('loader')).toBeTruthy();
  });

  test('Shouldreturn the small Loader', () => {
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Loader className="loader-small" />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
    expect(getByTestId('loader')).toBeTruthy();
  });
});
