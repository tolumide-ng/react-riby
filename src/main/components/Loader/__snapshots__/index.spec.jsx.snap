// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should return the Loader 1`] = `[Function]`;

exports[`Loader Shouldreturn the small Loader 1`] = `[Function]`;
